public/canvas_tools/tools.js

```diff
@@ -148,7 +152,7 @@ for (const tool of toolState.allTools) {
       "click",
       () => {
         toolState.curTool = tool;
```

Comment:
I started passing the core data to functions instead of passing the whole Javascript graph class which we will eventually remove